src/goto-harness/function_call_harness_generator.cpp

```diff
@@ -178,6 +181,11 @@ void function_call_harness_generatort::handle_option(
       }
     }
   }
+  else if(option == FUNCTION_HARNESS_GENERATOR_TREAT_POINTER_AS_CSTRING)
+  {
+    p_impl->function_parameters_to_treat_as_cstrings.insert(
```

Comment:
⛏ This option passing mechanism looks bad to me: You're creating an incomplete object and then setting fields after it's created. It sounds like these options should go into the constructor and maybe be `const` as you're not mutating these afterwards.

Doing so would remove the need for checks on these fields because there would be no wrong way to set them.

Reply:
1. there is no tangible benefit to the unordered variant in this case, and `set` is shorter.
2. the argument don't exist before we create them from the list of parameters. There is a map from parameters to arguments which we use to initialise the argument set though.
3. considered that, but would've made API more complicated in a different place so I discarded that idea